lib/checks/checks_test.go

```diff
@@ -118,41 +118,80 @@ func (s *ChecksSuite) TestTime(c *check.C) {
 }
 
 func (s *ChecksSuite) TestCheckSameOS(c *check.C) {
-	infos := []Server{
+	var testCases = []struct {
+		comment string
+		servers []Server
+		err     string
+	}{
 		{
-			ServerInfo: ServerInfo{
-				System: storage.NewSystemInfo(storage.SystemSpecV2{
-					Hostname: "node-1",
-					OS: storage.OSInfo{
-						ID:      "centos",
-						Version: "7.1",
-					},
-				}),
+			comment: "servers match on the major version",
+			servers: []Server{
+				newServer("node-1", "centos", "7.1"),
+				newServer("node-2", "centos", "7.2"),
+				newServer("node-3", "centos", "7.2"),
 			},
 		},
 		{
-			ServerInfo: ServerInfo{
-				System: storage.NewSystemInfo(storage.SystemSpecV2{
-					Hostname: "node-2",
-					OS: storage.OSInfo{
-						ID:      "centos",
-						Version: "7.2",
-					},
-				}),
+			comment: "servers do not match on the major version",
+			servers: []Server{
+				newServer("node-1", "centos", "7.1"),
+				newServer("node-2", "centos", "6.2"),
+				newServer("node-3", "centos", "7.2"),
 			},
+			err: "servers have different OS versions.*",
 		},
 		{
-			ServerInfo: ServerInfo{
-				System: storage.NewSystemInfo(storage.SystemSpecV2{
-					Hostname: "node-3",
-					OS: storage.OSInfo{
-						ID:      "centos",
-						Version: "7.2",
-					},
-				}),
+			comment: "match on the whole version",
+			servers: []Server{
+				newServer("node-1", "centos", "7"),
+				newServer("node-2", "centos", "7"),
 			},
 		},
+		{
+			comment: "no match on the whole version",
+			servers: []Server{
+				newServer("node-1", "centos", "7"),
+				newServer("node-2", "centos", "6"),
+			},
+			err: "servers have different OS versions.*",
+		},
+		{
+			comment: "no match on distribution",
+			servers: []Server{
+				newServer("node-1", "centos", "7"),
+				newServer("node-2", "rhel", "7"),
+			},
+			err: "servers have different OS distributions.*",
+		},
+		// {
+		// 	comment: "no match on bogus input? (identical to previous behavior)",
```

Comment:
Should this case be also supported? This seems unlikely as the `VERSION_ID` is [described](https://www.freedesktop.org/software/systemd/man/os-release.html#VERSION_ID=) as containing mostly numeric input.

Reply:
I like how the docs are precise, "mostly numeric" :) I think it's entirely possible system-release (or os-release) file can be malformed. But in this case it'll likely fail due to required version mismatch? So the OS "homogeneity" test doesn't necessarily need to account for it.